### ISSUE TITLE
dev_status and dangling pointer

### ISSUE THREAD

**Issue by jkotan** [tango-cs/pytango#24]
Hi,

When I override `dev_status` in PyTango 8 as

```
def dev_status(self):
    st = PyTango.Device_4Impl.dev_status(self)
    return st
```

and I call on the client side 

```
dp = PyTango.DeviceProxy("p09/pystate/01") 
dp.status()
```

I get

```
'Got exception\twhen trying to build device status'
```

However, the following code

```
def dev_status(self):
    self.st = PyTango.Device_4Impl.dev_status(self)
    return self.st
```

works OK.

Is there any reference counting problem?

Bests,
Jan

**Comment by vxgmichel** [comment]
Thanks for your report! It definitely look like a reference counting issue.
Is anyone willing to work on it?

**Comment by tiagocoutinho** [comment]
I have to apologize for this behavior. It was intentional to save an strdup and to avoid having to store the last status internally in PyTango Device class. 
The problem has already been discussed in the Tango mailing list and at the time I advise to follow the exact same recipe as @jkotan has in his second example.
I guess this is not expected behavior so I agree that it should be fixed.
I could work on this.

**Comment by tiagocoutinho** [comment]
It seems it is not fixed by #60.
It need some more work. A new PR will come soon
